src/gwt/src/org/rstudio/studio/client/workbench/views/source/editors/text/TextEditingTargetWidget.java

```diff
@@ -116,13 +124,88 @@ public void removeFindReplace(FindReplaceBar findReplaceBar)
             } 
          });
       
-      panel_ = new PanelWithToolbars(toolbar_ = createToolbar(fileType),
-                                    editor.asWidget(),
-                                    statusBar_);
+      editorPanel_ = new DockLayoutPanel(Unit.PX);
+      docOutlineWidget_ = new DocumentOutlineWidget(target);
+      
+      editorPanel_.addEast(docOutlineWidget_, 0);
+      editorPanel_.add(editor.asWidget());
+      
+      // Need to set the widget size deferred to ensure the docUpdateSentinel
+      // has been initialized as well
+      new Timer()
```

Comment:
I think we should make this relationship a bit tighter than just a millisecond delay (unless the delay itself is important for UI/appearance reasons). The DocUpdateSentinel is initialized in literally the next line of code after the TextEditingTargetWidget constructor. I definitely wouldn't change the order of construction, however you could do one of the following:

1. Add an explicit initWidgetSize call to TextEditingTargetWidget then call this in TextEditingTarget right after the DocUpdateSentinel is initialized (I think I like this option best because it documents the order dependency).

2. Do a scheduleFinally or rather than 100ms delay (since all you really need is to guarantee that the other code on the stack runs before you set the widget size). I don't like this alternative as much because it's less explicit.